functions/Copy-SqlSsisCatalog.ps1

```diff
@@ -0,0 +1,472 @@
+            throw "There was an error connecting to the source integration services."
+        }
+        try { 
+            Write-Verbose "Connecting to $Destination integration services."
+            $destinationSSIS = New-Object "$ISNamespace.IntegrationServices" $destinationConnection
+        }
+        catch { 
+            Write-Exception $_
+            throw "There was an error connecting to the destination integration services." 
+        }
+
+        $sourceCatalog = $sourceSSIS.Catalogs | Where-Object { $_.Name -eq "SSISDB" }  
+        $destinationCatalog = $destinationSSIS.Catalogs | Where-Object { $_.Name -eq "SSISDB" } 
+        
+        $sourceFolders = $sourceCatalog.Folders
+        $destinationFolders = $destinationCatalog.Folders
+    }
+    PROCESS
+    {
+        if (!$sourceCatalog) {
+            throw "The source SSISDB catalog does not exist."
+        }
+        if (!$destinationCatalog) {
+            if ($Pscmdlet.ShouldProcess($Destination, "Create destination SSISDB Catalog")) {
+                if (!$CreateCatalogPassword) {
+                    $message = "The destination SSISDB catalog does not exist, would you like to create one?"
+                    $yes = New-Object System.Management.Automation.Host.ChoiceDescription "&Yes", "Create an SSISDB catalog on $Destination."
+                    $no = New-Object System.Management.Automation.Host.ChoiceDescription "&No", "Exit."
+                    $options = [System.Management.Automation.Host.ChoiceDescription[]]($yes, $no)
+                    $result = $host.ui.PromptForChoice($null, $message, $options, 0) 
+                    switch ($result) {
+                        0 { New-SSISDBCatalog }
+                        1 { return }
+                    }
+                }
+                else {
+                    New-SSISDBCatalog -Password $CreateCatalogPassword
+                }
+
+                $destinationSSIS.Refresh()
+                $destinationCatalog = $destinationSSIS.Catalogs | Where-Object { $_.Name -eq "SSISDB" }
+                $destinationFolders = $destinationCatalog.Folders
+            }
+            else {
+                throw "The destination SSISDB catalog does not exist."
+            }
+        }
+        if ($folder) {
+            if ($sourceFolders.Name -contains $folder) {
+                $srcFolder = $sourceFolders | Where-Object { $_.Name -eq $folder }
+                if ($destinationFolders.Name -contains $folder) {
+                    if (!$force) {
+                        Write-Warning "Integration services catalog folder $folder exists at destination. Use -Force to drop and recreate."
+                    }
+                    else {
+                        if ($Pscmdlet.ShouldProcess($Destination, "Dropping folder $folder and recreating")) {
+                            try {
+                                New-CatalogFolder -Folder $srcFolder.Name -Description $srcFolder.Description -Force
```

Comment:
One of these commands is causing issues, i think it's this one

PS C:\github\branches> Copy-SqlSsisCatalog -Source sql2014 -Destination sql2016
WARNING: Integration services catalog folder foo exists at destination. Use -Force to drop and recreate.
Integration services migration finished.
PS C:\github\branches> Copy-SqlSsisCatalog -Source sql2014 -Destination sql2016 -Folder
Copy-SqlSsisCatalog : Missing an argument for parameter 'Folder'. Specify a parameter of type 'System.String' and try again.
At line:1 char:58
+ Copy-SqlSsisCatalog -Source sql2014 -Destination sql2016 -Folder